portal/src/graphql/portal/AuthenticationCountryCallingCodeList.tsx

```diff
@@ -0,0 +1,490 @@
+        case "selected":
+          return (
+            <CountryCallingCodeListItemCheckbox
+              index={index}
+              checked={item?.selected}
+              onCheckboxClicked={onCallingCodeSelected}
+            />
+          );
+        case "order":
+          if (item?.selected) {
+            return (
+              <OrderColumnButtons
+                index={index}
+                itemCount={selectedCountryCallingCodes.length}
+                onSwapClicked={onCallingCodeSwap}
+                renderAriaLabel={() => item.countryName}
+              />
+            );
+          }
+          return (
+            <span>
+              <FormattedMessage id="AuthenticationWidget.phone.default-order" />
+            </span>
+          );
+        case "countryName":
+          return <span>{item?.countryName}</span>;
+        case "callingCode":
+          return <span>{item?.callingCode}</span>;
+        default:
+          return <span>{item?.callingCode}</span>;
+      }
+    },
+    [onCallingCodeSwap, selectedCountryCallingCodes, onCallingCodeSelected]
+  );
+
+  const onRenderCallingCodeListHeader = useCallback<
+    IRenderFunction<IDetailsHeaderProps>
+  >(
+    (props) => {
+      if (props == null) {
+        return null;
+      }
+      const renderCheckbox = () => {
+        return (
+          <CountryCallingCodeListSelectAll
+            selectAll={selectAllCallingCode}
+            unselectAll={unselectAllCallingCode}
+            isPartiallySelected={isCallingCodePartiallySelected}
+            isAllSelected={isCallingCodeAllSelected}
+          />
+        );
+      };
+
+      // modify column width for select all checkbox
+      const modifiedColumns = produce(props.columns, (draftColumn) => {
+        const activeColumnWidth = draftColumn[0].calculatedWidth!;
+        draftColumn[0].calculatedWidth = activeColumnWidth - 35;
+      });
+
+      return (
+        <DetailsHeader
+          {...props}
+          columns={modifiedColumns}
+          onRenderDetailsCheckbox={renderCheckbox}
+          selectAllVisibility={SelectAllVisibility.visible}
+          styles={HEADER_STYLE}
+        />
+      );
+    },
+    [
+      selectAllCallingCode,
+      unselectAllCallingCode,
+      isCallingCodePartiallySelected,
+      isCallingCodeAllSelected,
+    ]
+  );
+
+  const onRenderCallingCodeListRow = useCallback<
+    IRenderFunction<IDetailsRowProps>
+  >(
+    (props) => {
+      if (props == null) {
+        return null;
+      }
+      const { itemIndex } = props;
+      const isLastSelectedRow =
+        itemIndex === selectedCountryCallingCodes.length - 1;
+      return (
+        <DetailsRow
+          {...props}
+          className={cn(styles.callingCodeListRow, {
+            [styles.lastSelectedCallingCode]: isLastSelectedRow,
+          })}
+        />
+      );
+    },
+    [selectedCountryCallingCodes]
+  );
+
+  return (
+    <section className={styles.root}>
+      <SearchBox
```

Comment:
For this kind of search box, it is more intuitive to search on text change.

Reply:
Sure, the performance is OK with search on text change.